agent/consul/discoverychain/gateway_test.go

```diff
@@ -557,8 +619,40 @@ func TestGatewayChainSynthesizer_Synthesize(t *testing.T) {
 									Partition: "default",
 									Namespace: "default",
 									RequestHeaders: &structs.HTTPHeaderModifiers{
-										Add: make(map[string]string),
-										Set: make(map[string]string),
+										Add: map[string]string{
+											"add me to the rule request":             "present",
+											"add me to the service request":          "present",
+											"add me to the rule and service request": "service",
```

Comment:
we expect `service` as the value here, indicating that the filter modifier on the service took precedence over the same modifier on the rule which specifies `rule` as the value.

this pattern applies to the other modifiers below as well.